pkg/commands/oscommands/os.go

```diff
@@ -173,15 +174,11 @@ func (c *OSCommand) FileExists(path string) (bool, error) {
 
 // PipeCommands runs a heap of commands and pipes their inputs/outputs together like A | B | C
 func (c *OSCommand) PipeCommands(commandStrings ...string) error {
-	cmds := make([]*exec.Cmd, len(commandStrings))
-	logCmdStr := ""
-	for i, str := range commandStrings {
-		if i > 0 {
-			logCmdStr += " | "
-		}
-		logCmdStr += str
-		cmds[i] = c.Cmd.New(str).GetCmd()
-	}
+	cmds := slices.Map(commandStrings, func(cmdString string) *exec.Cmd {
+		return c.Cmd.New(cmdString).GetCmd()
+	})
+
+	logCmdStr := strings.Join(commandStrings, " | ")
```

Comment:
I particularly like this example: the ceremony required to do a for loop can discourage devs from splitting a for loop into two when you want to handle two separate things. Having cmds and logCmdStr entangled makes it much harder to read the code because you start off assuming there's a dependency there. Of course, sometimes you need to just stick to one for loop for the sake of efficiency but I find that's rarely needed

Reply:
💯